src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -819,6 +822,11 @@ private ISymbolNode GetHelperFtnUncached(CorInfoHelpFunc ftnNum)
             return _compilation.NodeFactory.GetReadyToRunHelperCell(id);
         }
 
+        private CorInfoHelpFunc getLazyStringLiteralHelper(CORINFO_MODULE_STRUCT_* handle)
+        {
+            return CorInfoHelpFunc.CORINFO_HELP_STRCNS_CURRENT_MODULE;
```

Comment:
Is this right for strings from other modules? Crossgen1 version of this method has a check for current module in this method.

Reply:
I believe that's a fragile ngen scenario. `getLazyStringLiteralHelper` gets called from the below handling code in the JIT. Returning `CORINFO_HELP_STRCNS` causes the JIT to call `embedModuleHandle` which is not supported in R2R.
https://github.com/dotnet/runtime/blob/1390941e4f1076209b434bb4d0cda034ba9febdf/src/coreclr/jit/morph.cpp#L9287-L9301

Reply:
Does this all work well for large version bubble or composite mode cases?